src/components/NcContent/constants.ts

```diff
@@ -0,0 +1,11 @@
+import type { InjectionKey } from 'vue'
+
+/**
+ * Injects a callback to register the app navigation to the content container.
+ */
+export const INJECTION_KEY_HAS_APP_NAVIGATION: InjectionKey<(hasAppNavigation: boolean) => void> = Symbol.for('NcContent:setHasAppNavigation')
```

Comment:
More common name would be `HAS_APP_NAVIGATION_KEY`, to not duplicate the entire type name in the variable name.

Reply:
Or using it via composables, e.g. `useNcContentProvider()` and `useNcContent()`